### PR TITLE
Cancellable task

### PR DESCRIPTION
**Pull request by pfcoperez**
Added `Cancellable` abstraction whereby it is possible to build cancellable-like futures.

The underlying `Future` cancellation isn't controlled by the observers (`Future` instances) but by the mutable `Cancellable` thus keeping the immutable role of `Future`.

```scala
val ct = Cancellable[Int] {
    Thread.sleep(tsleep.toMillis)
     24
}

val f = ct.fut //This future is just an observer...
val f_meaning_life = ct.map(_*2) //... as well as this one

f_meaning_life.onFailure {
    case _: CancellationException => println("Not interested")
}

//We're not interested in the meaning of life anymore, lets free Deep Thought resources:
ct.cancel
```

### CODE REVIEW

File: src/main/scala/com/stratio/common/utils/concurrent/Cancellable.scala [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  val fut: Future[T] = Future {
+    jf.get()
+  }
+
+  /**
+    * Cancels the future job completing `fut` with `Failure(InterruptedException)`
+    */
+  def cancel(): Unit = jf.cancel(true)
+
+}
```
Comment (JSantosP) [inline src/main/scala/com/stratio/common/utils/concurrent/Cancellable.scala]: Not sure about the convention: CR at the end of file?

File: src/main/scala/com/stratio/common/utils/concurrent/Cancellable.scala [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    new Callable[T] {
+      override def call(): T = todo
+    }
+  )
+
+  executionContext.execute(jf)
+
+  implicit val _: ExecutionContext = executionContext
+
+  val fut: Future[T] = Future {
```
Comment (JSantosP) [inline src/main/scala/com/stratio/common/utils/concurrent/Cancellable.scala]: I think ```fut``` is not a well descriptive member for getting the future of the ```Callable```.

**Comment by JSantosP** [comment]
Apart from two dummy things, the rest of the PR seems awesome to me :-)

**Comment by dvallejo** [comment]
Alt Text

LGTM and thanks again for your work.

**Comment by JSantosP** [comment]
Now it looks cool to me :-)

**Pull request merged**